scipy/linalg/special_matrices.py

```diff
@@ -603,6 +603,41 @@ def companion(a):
     return c
 
 
+def helmert(n):
+    """
+    Create a Helmert matrix of order `n`.
+
+    This has applications in statistics, compositional or simplicial analysis,
+    and in Aitchison geometry.
+
+    Parameters
+    ----------
+    n : int
+        The size of the array to create.
+
+    Returns
+    -------
+    M : (n, n) ndarray
+        The Helmert matrix.
+
+    Examples
+    --------
+    >>> from scipy.linalg import helmert
+    >>> helmert(5)
+    array([[ 0.4472136 ,  0.4472136 ,  0.4472136 ,  0.4472136 ,  0.4472136 ],
+           [ 0.70710678, -0.70710678,  0.        ,  0.        ,  0.        ],
+           [ 0.40824829,  0.40824829, -0.81649658,  0.        ,  0.        ],
+           [ 0.28867513,  0.28867513,  0.28867513, -0.8660254 ,  0.        ],
+           [ 0.2236068 ,  0.2236068 ,  0.2236068 ,  0.2236068 , -0.89442719]])
+
+    """
+    H = np.tril(np.ones((n, n)), -1) - np.diag(np.arange(n))
+    d = np.arange(n) * np.arange(1, n+1)
+    H[0] = 1
+    d[0] = n
+    return H / np.sqrt(d)[:, np.newaxis]
```

Comment:
> But if it's for reference, then why is it in the code instead of in the docs?

I think most of the scipy special matrices are like this.  For example you probably never want to actually create the NxN dft matrix except for reference.  Instead, you want to use an equivalent N log N algorithm.  But one may want to use the matrix generation function to check that one's understanding or implementation of the N log N algorithm is correct.